webapp/publisher/snaps/views.py

```diff
@@ -677,6 +690,18 @@ def redirect_post_register_name():
 @publisher_snaps.route("/register-snap", methods=["POST"])
 @login_required
 def post_register_name():
+    try:
+        # We don't use the data from this endpoint.
+        # It is mostly used to make sure the user has signed
+        # the terms and conditions.
+        user = api.get_account(flask.session)
+    except ApiResponseErrorList as api_response_error_list:
+        return _handle_error_list(api_response_error_list.errors)
+    except ApiError as api_error:
+        return _handle_errors(api_error)
+
+    available_stores = logic.filter_available_stores(user["stores"])
```

Comment:
No need to run this function if the request works. Put it after line 707

Reply:
:+1: